packages/core/http/core-http-server/src/router/route.ts

```diff
@@ -321,6 +321,22 @@ export interface RouteConfigOptions<Method extends RouteMethod> {
    * @default false
    */
   httpResource?: boolean;
+
+  /**
+   * Based on the the ES API specification (see https://github.com/elastic/elasticsearch-specification)
+   * Kibana APIs can also specify some metadata about API availability.
+   *
+   * @remark intended to be used for informational purposes only.
+   *
+   */
+  availability?: {
```

Comment:
The ES specification has `availability` as either `stack` or `serverless` and we're missing that in the route config options.
It's probably not a deal breaker now but would be great to surface in the generated spec.